### DISCUSSION TITLE
InitialState in testing does not work correctly

### DISCUSSION THREAD

**Discussion by Xeyos88** [vuejs/pinia#2524] [Help and Questions] [answered]
Hi,
I have a problem with initialising the status of a store in tests.

My store:
```
export const useUserStore = defineStore({
  id: 'user',
  state: () => ({
    profile: ref<JSONValue>(),
    role: 'user' as accessKey
  }),
...
```

My component:
```
...
const userStore = useUserStore()
const { profile, role } = storeToRefs(userStore)
....

<span>Hi, {{ profile.preferred_username }}</span>
```

Test:
```
describe('Test Layout View', () => {
  const wrapper = mount(LayoutView, {
    global: {
      plugins: [
        createTestingPinia({
          initialState: {
            user: {
              profile: {
                preferred_username: 'etopa'
              },
              role: 'admin'
            }
          }
        })
      ],
```
It seems to me that the status is initialised correctly and yet I receive this error:

> Cannot read properties of undefined (reading 'preferred_username')

I have the exact same test in other projects but it does not seem to work here.

I have no major differences except for the package manager which here is pnpm. Same version of pnpm, pina/testing and vitest

**Comment by posva** [comment]
The only thing I would change is `profile: null as JSONValue | null`. You don't need a ref there. Other than that, the error could come from a different test?

**Comment by Xeyos88** [comment, marked as answer]
Hi, 
I forgot to update the answer.
The realte error was not on pinia-testing.
In the same page I import the library 'js-sha256' which must be mocked otherwise this creates problems.
Vitest, however, incorrectly reported the error on the store part
